packages/medusa/src/loaders/database.ts

```diff
@@ -1,48 +1,61 @@
 import {
-  Connection,
-  createConnection,
-  getConnectionManager,
-  getConnection,
+  DataSource,
+  DataSourceOptions,
+  Repository,
+  TreeRepository,
 } from "typeorm"
-import { ShortenedNamingStrategy } from "../utils/naming-strategy"
 import { AwilixContainer } from "awilix"
-import { ConnectionOptions } from "typeorm/connection/ConnectionOptions"
 import { ConfigModule } from "../types/global"
+import "../utils/naming-strategy"
 
 type Options = {
   configModule: ConfigModule
   container: AwilixContainer
+  customOptions?: {
+    migrations: DataSourceOptions["migrations"]
+    logging: DataSourceOptions["logging"]
+  }
+}
+
+export let dataSource: DataSource
+
+if (process.env.NODE_ENV === "test") {
+  dataSource = {
+    getRepository: (target) => new Repository(target, {} as any) as any,
+    getTreeRepository: (target) => new TreeRepository(target, {} as any) as any,
+  } as unknown as DataSource
```

Comment:
nit(non-blocking): Can you add a comment as to why we do this? It is not super clear, why these are mocked here and in this way.

Reply:
Bump

Reply:
Changing this to a todo, as I don't find it super clear, why this is needed for a test environment

Reply:
I'm not sure why this was done either. @adrien2p do you?

Reply:
I think we can remove it now, it was temporary in my previous PR to mock those methods

Reply:
the idea was to not have to change all the integration test, but ill do it

Reply:
@olivermrbl if that is fine, we can keep it for later. The issue (now I remember) is how the files are imported. In the integration tests, when the file is loaded the import are evaluated and retrieved, since they import from medusa, the repository are part of the exports, therefore they are loaded as well. But at that moment the datasource in the server does not exists yet at it has not been loaded yet